FreeRTOS/FreeRTOS-Plus-TCP
# Add IPv6 specific header files.

Description
-----------

- Split FreeRTOS_IP header files to IPv4 and IPv6 specific header files and add IPv6 specific structures.

> FreeRTOS_IPv4.h
> FreeRTOS_IPv4_Private.h
> FreeRTOS_IPv6.h
> FreeRTOS_IPv6_Private.h
> FreeRTOS_IP_Common.h

- Added header files based on ipconfigUSE_IPV4 and ipconfigUSE_IPV6 to ensure backward compatibility.

- Created a common IP address structure to handle both IPv4 and IPv6 address : 
      IP_Address_t xIPAddress;      /**< Source or destination IP address, depending on usage scenario. */


Test Steps
-----------
As of now compilation test is being done. *
```
cmake -S test/build-combination -B test/build-combination/build/ -DTEST_CONFIGURATION=ENABLE_ALL
make -C test/build-combination/build/
```

Related Issue
-----------
As unit tests will be done after merging other changes as well. **It might break some test cases as of now which will be fixed in upcoming PRs.**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
